### PR TITLE
Wazuh changes to fit the ruleset normalization to Wazuh standard - epic implementation

### PR DESCRIPTION
**Pull request by DProvinciani**
|Related issue|
|---|
| #6866 |

## Description

This PR includes all the necessary changes in the Wazuh code in order to match with the changes introduced in the ruleset as part of the epic: [Ruleset normalization to Wazuh standard](https://github.com/wazuh/wazuh-ruleset/issues/784).

### Changes included

- Logs modifications to match with the new wazuh decoders and rules.
- Rename of the `OSSECAlert_Decoder` to `WazuhAlert_Decoder`.
- Update of all the ruleset in the `wazuh` repository.

### Testing done

The testing done combines the execution of the ruleset testing project and exploratory testing. For details, check the testing section in the pull https://github.com/wazuh/wazuh-ruleset/pull/812.

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
  - [x] Windows
  - [x] MAC OS X
- [x] Source installation
- [x] Package installation
- [x] Source upgrade
- [x] Package upgrade
- [x] Review logs syntax and correct language

- Memory tests for Linux
  - [x] Scan-build report
  - [x] Coverity

### CODE REVIEW

File: api/api/spec/spec.yaml [hunk 1]
```diff
@@ -10991,13 +10991,13 @@ paths:
...
-                    - file: 0015-ossec_rules.xml
+                    - file: 0015-wazuh_rules.xml
                       path: ruleset/rules
                       status: enabled
-                    - file: 0016-wazuh_rules.xml
+                    - file: 0020-syslog_rules.xml
                       path: ruleset/rules
                       status: enabled
-                    - file: 0020-syslog_rules.xml
+                    - file: 0025-sendmail_rules.xml
```
Comment (pereyra-m) [inline api/api/spec/spec.yaml:11000]: We should check the `0025-sendmail_rules.xml`, is this change related to this Epic ?

File: api/test/integration/env/configurations/rbac/rule/black_config.yaml [hunk 2]
```diff
@@ -3,6 +3,5 @@
   - rules:read
   resources:
   - rule:file:0010-rules_config.xml
-  - rule:file:0015-ossec_rules.xml
```
Comment (pereyra-m) [inline api/test/integration/env/configurations/rbac/rule/black_config.yaml:6]: Are we sure about this removal? Shouldn't `0015-wazuh_rules.xml` go here?

File: api/test/integration/env/configurations/rbac/rule/white_config.txt [hunk 3]
```diff
@@ -1,8 +1,8 @@
 | Resource  | List                                                         |
 |-----------|--------------------------------------------------------------|
-| rule:file | 0010-rules_config.xml 0015-ossec_rules.xml local_rules.xml   |
+| rule:file | 0010-rules_config.xml 0015-wazuh_rules.xml local_rules.xml   |
 |           | 0610-win-ms_logs_rules.xml 0620-win-generic_rules.xml ...    |
 
 | Actions    | Resources | Allow                                      | Deny |
 |------------|-----------|--------------------------------------------|------|
-| rules:read | rule:file | 0010-rules_config.xml 0015-ossec_rules.xml |      |
+| rules:read | rule:file | 0010-rules_config.xml                      |      |
```
Comment (pereyra-m) [inline api/test/integration/env/configurations/rbac/rule/white_config.txt:8]: Are we sure about this removal? Shouldn't `0015-wazuh_rules.xml` go here?

File: api/test/integration/env/configurations/rbac/rule/white_config.yaml [hunk 4]
```diff
@@ -3,5 +3,4 @@
   - rules:read
   resources:
   - rule:file:0010-rules_config.xml
-  - rule:file:0015-ossec_rules.xml
```
Comment (pereyra-m) [inline api/test/integration/env/configurations/rbac/rule/white_config.yaml:6]: Are we sure about this removal? Shouldn't `0015-wazuh_rules.xml` go here?

File: src/logcollector/logcollector.c [hunk 5]
```diff
@@ -651,12 +651,12 @@ void LogCollectorStart()
...
                         char msg_alert[512 + 1];
 
-                        snprintf(msg_alert, 512, "ossec: File rotated (inode "
+                        snprintf(msg_alert, 512, "wazuh: File rotated (inode "
                                  "changed): '%s'.",
                                  current->file);
 
                         /* Send message about log rotated */
-                        w_msg_hash_queues_push(msg_alert, "wazuh-logcollector", strlen(msg_alert) + 1, default_target, LOCALFILE_MQ);
+                        w_msg_hash_queues_push(msg_alert, "ossec-logcollector", strlen(msg_alert) + 1, default_target, LOCALFILE_MQ);
```
Comment (pereyra-m) [inline src/logcollector/logcollector.c:659]: Here, we are replacing "**wazuh**-logcollector" by "**ossec**-logcollector"

File: src/logcollector/logcollector.c [hunk 6]
```diff
@@ -682,12 +682,12 @@ void LogCollectorStart()
...
                         char msg_alert[512 + 1];
 
-                        snprintf(msg_alert, 512, "ossec: File size reduced "
+                        snprintf(msg_alert, 512, "wazuh: File size reduced "
                                  "(inode remained): '%s'.",
                                  current->file);
 
                         /* Send message about log rotated */
-                        w_msg_hash_queues_push(msg_alert, "wazuh-logcollector", strlen(msg_alert) + 1, default_target, LOCALFILE_MQ);
+                        w_msg_hash_queues_push(msg_alert, "ossec-logcollector", strlen(msg_alert) + 1, default_target, LOCALFILE_MQ);
```
Comment (pereyra-m) [inline src/logcollector/logcollector.c:690]: Here, we are replacing "**wazuh**-logcollector" by "**ossec**-logcollector"

File: api/api/spec/spec.yaml (hunk 1 above)
Comment (palaciosjeremias) [reply api/api/spec/spec.yaml:11000]: This change is part of [#6831](https://github.com/wazuh/wazuh/pull/6831), and was validated by the framework team.

File: api/test/integration/env/configurations/rbac/rule/black_config.yaml (hunk 2 above)
Comment (palaciosjeremias) [reply api/test/integration/env/configurations/rbac/rule/black_config.yaml:6]: This change is part of [#6831](https://github.com/wazuh/wazuh/pull/6831) and was validated by the framework team.

File: api/test/integration/env/configurations/rbac/rule/white_config.txt (hunk 3 above)
Comment (palaciosjeremias) [reply api/test/integration/env/configurations/rbac/rule/white_config.txt:8]: This change is part of [#6831](https://github.com/wazuh/wazuh/pull/6831) and was validated by the framework team.

File: api/test/integration/env/configurations/rbac/rule/white_config.yaml (hunk 4 above)
Comment (palaciosjeremias) [reply api/test/integration/env/configurations/rbac/rule/white_config.yaml:6]: This change is part of [#6831](https://github.com/wazuh/wazuh/pull/6831) and was validated by the framework team.

File: src/logcollector/logcollector.c (hunk 5 above)
Comment (palaciosjeremias) [reply src/logcollector/logcollector.c:659]: It's true! This was a rebase error. Nice catch.

File: src/logcollector/logcollector.c (hunk 6 above)
Comment (palaciosjeremias) [reply src/logcollector/logcollector.c:690]: It's true! This was a rebase error. Nice catch.

**Review by pereyra-m — APPROVED** [review]
LGTM!

**Comment by palaciosjeremias** [comment]
Successful execution of **runtest.py**:
Screenshot from 2021-02-03 15-46-55
Screenshot from 2021-02-03 15-47-30

File: framework/wazuh/core/tests/test_rule.py [hunk 7]
```diff
@@ -29,7 +29,7 @@
 
 ruleset_conf = {
     'decoder_dir': ['ruleset/decoders', 'etc/decoders'],
-    'rule_dir': ['ruleset/rules', 'etc/rules'], 'rule_exclude': ['0215-policy_rules.xml'],
+    'rule_dir': ['framework/wazuh/core/tests/data/rules'], 'rule_exclude': ['0215-policy_rules.xml'],
```
Comment (palaciosjeremias) [inline framework/wazuh/core/tests/test_rule.py:32]: Is this path, and the format of the array correct?
**decoder_dir**, i.e., uses two elements for the array and target to the folders where the rules/decoders are originally placed.

**Review by davidjiglesias — COMMENTED** [review]
I see some Copyright changes to Copyright (C) 2015-2020, it should be Copyright (C) 2015-2021 already. Rest LGTM!

File: framework/wazuh/core/tests/test_rule.py (hunk 7 above)
Comment (davidjiglesias) [reply framework/wazuh/core/tests/test_rule.py:32]: All ok, the test passed correctly

**Comment by pereyra-m** [comment]
Also, the rules UT were run with `runtests.py` 
https://ci.wazuh.info/job/test_ruleset/274/console

**Review by vikman90 — APPROVED** [review]
